src/main/java/org/fcrepo/auth/webac/LinkHeaderProvider.java

```diff
@@ -69,15 +73,20 @@
         final ListMultimap<String, String> headers = ArrayListMultimap.create();
 
         LOGGER.debug("Adding WebAC Link Header for Resource: {}", resource);
-
-        nodeService.find(internalSession, resource.getPath()).getTriples(translator, PropertiesRdfContext.class)
-        .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
-        .filter(t -> t.getObject().isURI())
-        .forEachRemaining(t -> {
-                    headers.put("Link", Link.fromUri(uriInfo.getBaseUriBuilder()
-                            .path(translator.convert(model.asStatement(t).getObject().asResource()).getPath())
-                            .toString()).rel("acl").build().toString());
-        });
+        // Get the correct Acl for this resource
+        final Optional<Pair<URI, FedoraResource>> acls = WebACRolesProvider.getEffectiveAcl(resource);
+        if (acls.isPresent()) {
+            // If the Acl is present we probably need to use the internal session to get it's URI
+            nodeService.find(internalSession, acls.get().getRight().getPath())
+                    .getTriples(translator, PropertiesRdfContext.class)
+                    .filter(t -> model.asStatement(t).getPredicate().hasURI(WEBAC_ACCESS_CONTROL_VALUE))
+                    .filter(t -> t.getObject().isURI())
```

Comment:
My concern with that approach is the exception appears at the point of data access, not at data submission. So a PUT/POST/PATCH with incorrect data would succeed but the GET request would fail.